eclass/kernel-install.eclass

```diff
@@ -766,13 +766,33 @@ kernel-install_compress_modules() {
 
 	if use modules-compress; then
 		einfo "Compressing kernel modules ..."
-		# xz options taken from scripts/Makefile.modinst
-		# we don't do 'xz -T' because it applies multithreading per file,
-		# so it works only for big files, and we have lots of small files
-		# instead
-		find "${ED}/lib" -name '*.ko' -print0 |
-			xargs -0 -P "$(makeopts_jobs)" -n 128 \
-				xz --check=crc32 --lzma2=dict=1MiB
+		if [[ -z ${KV_FULL} ]]; then
+			KV_FULL=${PV}${KV_LOCALVERSION}
+		fi
+		local compress=$(dist-kernel_detect_compressor "${ED}/usr/src/linux-${KV_FULL}")
+		local compress_args=()
+		# Options taken from linux-mod-r1.eclass.
+		# We don't instruct the compressor to parallelize because it applies
+		# multithreading per file, so it works only for big files, and we have
+		# lots of small files instead.
+		if [[ ${compress} == .ko ]]; then
+			# Nothing to do
+			return
+		elif [[ ${compress} == .ko.gz ]]; then
+			if type -P pigz &>/dev/null; then
+				compress_args+=( pigz )
```

Comment:
We don't want parallel compressors, we compress multiple modules in parallel. As the comment above it says.